server/src/api/UserInterestRoutes.ts

```diff
@@ -0,0 +1,134 @@
+import express from "express";
+import { authenticateToken } from "../middlewares";
+import { prisma } from "../lib/prisma";
+import { INTERNAL_ERROR_CODE, INVALID_REQUEST_CODE, OK_CODE } from "../lib/StatusCodes";
+import { Category, UserInterest } from "@prisma/client";
+
+const router = express.Router();
+
+type InterestsResponse = {
+  faculties: Category[],
+  ethnicities: Category[],
+  religions: Category[],
+  programs: Category[],
+  hobbies: Category[]
+}
+
+type UserInterestsRequest = {
+  faculty: number,
+  ethnicity: number,
+  religion: number,
+  program: number,
+  hobbies: number[]
+}
+
+type UserInterestsResponse = {
+  interests: UserInterest[]
+}
+
+type UserInterestWithoutId = {
+  user_id: number,
+  category_id: number
+}
+
+router.get<void, InterestsResponse>("/all", authenticateToken, async (req, res) => {
+  try {
+    const promises = [
+      prisma.category.findMany({
```

Comment:
just get all in a single async call and do local group logic.

Reply:
resolved.